src/librustc_trans/mir/block.rs

```diff
@@ -115,6 +115,9 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                 if let Some(cleanup_pad) = cleanup_pad {
                     bcx.cleanup_ret(cleanup_pad, None);
                 } else {
+                    let llpersonality = bcx.fcx.eh_personality();
```

Comment:
I'm looking into that now. The issue that we found in avr-rust/rust#18 was having a `resume` without a personality inside `std::mem::forget`, which is basically a `nop` so I don't think there is any cleanup to do.